### ISSUE TITLE
Output from bindgen doesn't get included in build tree

### ISSUE THREAD

**Issue by MerlinDE** [intellij-rust/intellij-rust#6135]
## Environment

* **IntelliJ Rust plugin version:** 0.3.130.3337-202
* **Rust toolchain version:** 1.48.0-nightly (bbc677480 2020-09-18) x86_64-apple-darwin
* **IDE name and version:** CLion 2020.2.3 (CL-202.7319.72)
* **Operating system:** macOS 10.15.6

## Problem description

When using `bindgen`or similar crate, that generates Rust code at build time for inclusion in the project, that generated code doesn't become part of the build tree. That means I can't jump to functions declared there and also have to manually search for the generated `.rs` file in the `target` folder.

As the output is included in the overall project via `include!(concat!(env!("OUT_DIR"), "/generated-bindings-sys.rs"));` it should be available for analysis and reference inside the IDE.

**Comment by Undin** [comment]
@MerlinDE By default, the plugin doesn't know location of `OUT_DIR` env variable value. It leads to unknown name for name resolution mechanism.
Try to enable `org.rust.cargo.evaluate.build.scripts` experimental feature (`Help | Find Action` and type `Experimental Features`). After that, each import of project data from Cargo (aka Cargo refresh) will launch `cargo check` to extract `OUT_DIR` for each crate. See #4734 for more details
Note it may significantly increase time of Cargo refresh operation

**Comment by MerlinDE** [comment]
@Undin Awesome! That helps a lot. In addition to `org.rust.cargo.evaluate.build.scripts`that you've mentioned, I also found `org.rust.cargo.fetch.out.dir` which seems to also handle that case. I wonder if those two options work together or can also be used on their own with may be less impact on refresh time?

**Comment by Undin** [comment]
@MerlinDE yep, `org.rust.cargo.fetch.out.dir` can also help but it has some major disadvantages:
- breaks incremental compilation
- helps only with `OUT_DIR`
- uses unstable `build-plan` command that supposed to be removed (see https://github.com/rust-lang/cargo/issues/7614 for more details)

At the same time, `org.rust.cargo.evaluate.build.scripts` can handle other items generated by build scripts like `cfg` attributes and environment variables and doesn't break incremental compilation. Yeah, it launches `cargo check` under the hood that can be quite long the first time but due to incremental compilation, subsequent launches will be much faster.

> I wonder if those two options work together 

If `org.rust.cargo.evaluate.build.scripts` is enabled, `org.rust.cargo.fetch.out.dir` will be ignored.

> or can also be used on their own with may be less impact on refresh time?

I wouldn't recommend using `org.rust.cargo.fetch.out.dir` instead of `org.rust.cargo.evaluate.build.scripts` because it breaks incremental compilation and may cause regression in your development performance (especially if your project is quite big).  This option exists because of historical reasons (it appeared earlier) and I think it will be removed during implementation of #6104
